### PR TITLE
Companion for #11649: Bound uses of `Call`

### PR DESCRIPTION
**Pull request by gavofyork**
Companion to https://github.com/paritytech/substrate/pull/11649. This MR contains 15 migrations in total.

It migrates the following pallets in that exact order:
- Preimages (DOT, KSM, ROC, WND)
- Scheduler (DOT, KSM, ROC, WND)
- Democracy (DOT, KSM, ROC)
- Multisig (DOT, KSM, ROC, WND) https://github.com/paritytech/substrate/pull/12072

TODO:
- [x] Add migration for Preimage pallet (Kusama, Westend, Polkadot, Rococo)
- [x] Add migration for Scheduler pallet (Kusama, Westend, Polkadot, Rococo)
- [x] Add democracy for Scheduler pallet (Kusama, Westend, Polkadot, Rococo)
- [ ] Try-runtime testing
  - [x] Polkadot: [comment](https://github.com/paritytech/polkadot/pull/5729#issuecomment-1265620778) and [CI-log](https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/1909945)
  - [x] Kusama: [comment](https://github.com/paritytech/polkadot/pull/5729#issuecomment-1265738204) and [CI-log](https://gitlab.parity.io/parity/mirrors/polkadot/-/jobs/1910404). The migration will drop 3 un-decodable agendas.
- [x] Update weights

### CODE REVIEW

**Review by shawntabrizi — APPROVED** [review]

**Review by rdovgan — APPROVED** [review]

**Comment by ggwpez** [comment]
/cmd queue -v RUST_LOG=remote-ext=debug,runtime=trace -c try-runtime $ --chain=polkadot-dev --execution=Wasm --no-spec-name-check on-runtime-upgrade live --uri wss://polkadot-try-runtime-node.parity-chains.parity.io:443

**Comment by ggwpez** [comment]
/cmd queue -v PATCH_substrate=11649 -v RUST_LOG=remote-ext=debug,runtime=trace -c try-runtime $ --chain=polkadot-dev --execution=Wasm --no-spec-name-check on-runtime-upgrade live --uri wss://polkadot-try-runtime-node.parity-chains.parity.io:443

**Comment by ggwpez** [comment]
/cmd queue -v PATCH_substrate=11649 -v RUST_LOG=remote-ext=debug,runtime=trace -c try-runtime $ --chain=polkadot-dev --execution=Wasm --no-spec-name-check on-runtime-upgrade live --uri wss://polkadot-try-runtime-node.parity-chains.parity.io:443

**Comment by ggwpez** [comment]
/cmd queue -v PATCH_substrate=11649 -v RUST_LOG=remote-ext=debug,runtime=trace -c try-runtime $ --chain=polkadot-dev --execution=Wasm --no-spec-check-panic on-runtime-upgrade live --uri wss://polkadot-try-runtime-node.parity-chains.parity.io:443

**Comment by ggwpez** [comment]
The try-runtime nodes time-out again. Not sure what is going on. I asked Devops.

**Comment by ggwpez** [comment]
/cmd queue -v PATCH_substrate=11649 -v RUST_LOG=remote-ext=debug,runtime=trace -c try-runtime $ --chain=polkadot-dev --execution=Wasm --no-spec-check-panic on-runtime-upgrade live --uri wss://polkadot-try-runtime-node.parity-chains.parity.io:443

**Comment by ruseinov** [comment]
@ggwpez I'm branching off that to introduce a migration and try-runtime test it.

**Comment by ruseinov** [comment]
bot rebase

**Comment by ggwpez** [comment]
bot rebase

**Comment by ggwpez** [comment]
/cmd queue -v PATCH_substrate=11649 -v RUST_LOG=remote-ext=debug,runtime=trace -c try-runtime $ --chain=polkadot-dev --execution=Wasm --no-spec-check-panic on-runtime-upgrade live --uri wss://polkadot-try-runtime-node.parity-chains.parity.io:443

**Comment by ggwpez** [comment]
/cmd queue -v PATCH_substrate=11649 -v RUST_LOG=remote-ext=debug,runtime=trace -c try-runtime $ --chain=polkadot-dev --execution=Wasm --no-spec-check-panic on-runtime-upgrade live --uri ws://try-runtime.tasty.limo:9944

**Comment by ggwpez** [comment]
bot rebase

**Comment by ggwpez** [comment]
/cmd queue -v PATCH_substrate=11649 -v RUST_LOG=remote-ext=debug,runtime=trace -c try-runtime $ --chain=polkadot-dev --execution=Wasm --no-spec-check-panic on-runtime-upgrade live --uri ws://try-runtime.tasty.limo:9944

**Comment by ggwpez** [comment]
/cmd queue -v PATCH_substrate=11649 -v RUST_LOG=remote-ext=debug,runtime=trace -c try-runtime $ --chain=kusama-dev --execution=Wasm --no-spec-check-panic on-runtime-upgrade live --uri ws://try-runtime.tasty.limo:9944

**Comment by ggwpez** [comment]
/cmd queue -c bench-bot $ pallet kusama-dev pallet_democracy

File: runtime/westend/src/lib.rs [hunk 1]
```diff
@@ -1215,6 +1213,9 @@ pub type Executive = frame_executive::Executive<
 			StakingMigrationV11OldPallet,
 		>,
 		pallet_staking::migrations::v12::MigrateToV12<Runtime>,
+		// "Bound uses of call" <https://github.com/paritytech/polkadot/pull/5729>
+		pallet_preimage::migration::v1::Migration<Runtime>,
+		pallet_scheduler::migration::v3::MigrateToV4<Runtime>,
```
Comment (ggwpez) [inline runtime/westend/src/lib.rs:1218]: Multisig migration missing here and in Rococo.

**Comment by gavofyork** [comment]
bot merge

**Comment by ggwpez** [comment]
The failing try-runtime is still slow nodes. They are much faster than formerly, but something is still not quite right.  
related https://github.com/paritytech/ci_cd/issues/573 and https://github.com/paritytech/devops/issues/2013

**Review by ordian — APPROVED** [review]

**Comment by ordian** [comment]
bot merge

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis/1026/2

**Pull request merged**